runtime/src/asset.rs

```diff
@@ -157,6 +179,70 @@ decl_module! {
         /// initialize the default event for this module
         fn deposit_event() = default;
 
+        /// This function is used to either register a new ticker or extend validity of an exisitng ticker
+        /// NB Ticker validity does not get carryforward when renewing ticker
+        ///
+        /// # Arguments
+        /// * `origin` It consist the signing key of the caller (i.e who signed the transaction to execute this function)
+        /// * `to_did` DID of the (future) owner of the ticker
+        /// * `_ticker` ticker to register
+        pub fn register_ticker(origin, to_did: IdentityId, _ticker: Vec<u8>) -> Result {
+            let _ = ensure_signed(origin)?;
+
+            let ticker = utils::bytes_to_upper(_ticker.as_slice());
+
+            ensure!(!<Tokens<T>>::exists(&ticker), "token already created");
+
+            let ticker_config = Self::ticker_registration_config();
+
+            ensure!(ticker.len() <= usize::try_from(ticker_config.max_ticker_length).unwrap_or_default(), "ticker length over the limit");
+
+            if Self::is_ticker_available(&ticker) {
+                // ticker not registered by anyone (or registry expired). we can charge fee and register this ticker
+            } else {
+                // ticker already registered to someone. Ensure that the ticker is registered to same did
+                ensure!(Self::is_ticker_registry_valid(&ticker, to_did), "ticker registered to someone else");
```

Comment:
Does ensure! do lazy evaluation, e.g. would something like:
`ensure!(Self::is_ticker_available(&ticker) || Self::is_ticker_registry_valid(&ticker, to_did))`
be more concise and as efficient?

Reply:
I think it does. The reason I stored the results in variable is that we need them later. However, we only need the result of `is_ticker_available` and not `is_ticker_registry_valid` so we can store that result and still lazy evaluate the other one. 

In fact, both these functions have some common code and we can create a new function that has less redundant code. Thanks for pointing out, I'll push an update in a bit.